lib/opal/nodes/top.rb

```diff
@@ -19,11 +19,11 @@ def compile
           body_code = [body_code] unless body_code.is_a?(Array)
 
           if compiler.eval?
-            add_temp '$scope = (self.$$scope || self.$$class.$$scope)'
+            add_temp '$nesting = self.$$is_a_module ? [self] : [self.$$class]'
```

Comment:
@meh there's `$$is_module` and `$$is_class` that XOR on modules and classes, but there's also `$$is_a_module` that is `true` for both and replaces `$$is_module || $$is_class` checks. 

That said I agree that a better name is desirable.